services/121-service/src/payments/fsp-integration/intersolve-visa/intersolve-visa.service.ts

```diff
@@ -1,1453 +1,779 @@
+        .intersolveVisaChildWallets[0];
 
     try {
-      const relationOptions = await this.getRelationOptionsForVisa(referenceId);
-      const paymentDetails =
-        await this.registrationDataQueryService.getPaDetails(
-          [referenceId],
-          relationOptions,
-        );
-
-      const addressPayload = this.createCustomerAddressPayload(
-        paymentDetails[0],
+      await this.executeReissueCardSteps(
+        input,
+        intersolveVisaCustomer,
+        childWalletToReplace,
       );
-      const addressResult =
-        await this.intersolveVisaApiService.updateCustomerAddress(
-          visaCustomer?.holderId ?? null,
-          addressPayload,
-        );
-      if (!this.isSuccessResponseStatus(addressResult.status)) {
-        errors.push(`Address update failed: ${addressResult?.data?.code}.`);
-      }
-
-      if (errors.length > 0) {
-        throw new HttpException({ errors }, HttpStatus.INTERNAL_SERVER_ERROR);
-      }
     } catch (error) {
-      if (error.name === ErrorEnum.RegistrationDataError) {
-        console.info(
-          `Unable to sync address data because this registration does not have this data anymore.\n
-          This is most likely because this registration first had the FSP Intersolve Visa, and then switched to another FSP\n
-          This new fsp does not have the attributes needed for Intersolve Visa, so the data is removed from the registration`,
+      if (error instanceof IntersolveVisaApiError) {
+        throw new HttpException(
+          `${IntersolveVisa121ErrorText.reissueCard} - ${error.message}`,
+          HttpStatus.NOT_FOUND,
         );
       } else {
         throw error;
       }
     }
   }
 
-  public async reissueWalletAndCard(
-    referenceId: string,
-    programId: number,
-    userId: number,
-  ): Promise<any> {
-    const { registration: _registration, visaCustomer } =
-      await this.getRegistrationAndVisaCustomer(referenceId, programId);
-    if (visaCustomer.visaWallets.length === 0) {
-      const errors = `No wallet available yet for PA with this referenceId ${referenceId}`;
-      throw new HttpException({ errors }, HttpStatus.NOT_FOUND);
-    }
-
-    const oldWallets = visaCustomer.visaWallets.sort((a, b) =>
-      a.created < b.created ? 1 : -1,
-    );
-    const oldWallet = oldWallets[0];
-
-    const errorGenericPart = `<br><br>Update data if applicable and retry by using the 'Issue new card' button again. If the problem persists, contact 121 technical support.`;
-    // 0. sync customer data with 121 data, as create-customer is skipped in this flow
-    try {
-      await this.syncIntersolveCustomerWith121(referenceId, programId);
-    } catch (error) {
-      const errors = `SYNC CUSTOMER DATA ERROR: <strong>${error.response?.errors.join(
-        ',',
-      )}</strong>${errorGenericPart}`;
-      throw new HttpException({ errors }, HttpStatus.INTERNAL_SERVER_ERROR);
+  public async executeReissueCardSteps(
```

Comment:
@PeterSmallenbroek Did you create this function? Git blame thinks it was me, but I think not.

I think this goes against our guideline of not "passing through" param structures from one function to the next. In this case: ReissueCardParams. 

I suggest to refactor this into the way that doTransferOrIssueCard was refactored into smaller private methods.

What do you think?

Reply:
This was created by @RubenGeo 
I think the contents of the 'executeReissueCardSteps' method can be moved to the try{} block already. So, I agree that it can use some improving.